openstack/train/ovs-kernel/nova.patch

```diff
@@ -106,36 +158,59 @@ index f4df91e..09585ca 100644
      def _get_guest_config_meta(self, instance):
          """Get metadata config for guest."""
  
-@@ -5821,12 +5828,32 @@ class LibvirtDriver(driver.ComputeDriver):
-         for config in storage_configs:
-             guest.add_device(config)
- 
-+        net_index = 0
-         for vif in network_info:
-             config = self.vif_driver.get_config(
+@@ -5826,6 +5833,16 @@ class LibvirtDriver(driver.ComputeDriver):
                  instance, vif, image_meta,
                  flavor, virt_type, self._host)
              guest.add_device(config)
- 
-+            if vif.get("vnic_type") == "virtio-forwarder":
-+                # Enable page-per-vq for virtio-forwarder interface
-+                args_dict = {"page-per-vq": "on"}
++            if vif.get("vnic_type") == network_model.VNIC_TYPE_VIRTIO_FORWARDER:
 +                if (vif.get("network") and vif.get("network").get("meta") and
 +                        vif.get("network").get("meta").get("mtu")):
 +
 +                    # As a workaround, we have to set mtu_host in xml file for
 +                    # virtio-forwarder interface, we may revist it if needed
-+                    args_dict["host_mtu"] = vif["network"]["meta"]["mtu"]
-+                qemu_args = self._get_guest_qemu_commandline("net%d"
-+                                                             % net_index,
-+                                                             **args_dict)
-+                guest.qemu_args.append(qemu_args)
-+            if vif.get("vnic_type") in ["virtio-forwarder", "normal", "macvtap"]:
-+                # As the alias name of virtio-forwarder, macvtap and normal
```

Comment:
why we alisa virtio-forwarder, macvtap and normal and not just virtio-forwarder?